### PR TITLE
fix: Resolve existing type issues

### PR DESCRIPTION
**Pull request by xeho91**
## Things to note

- Removed first generic type parameter `TOverrideArgs` which was incomplete.
  I believe is better to focus on getting types to work correctly than adding a new incomplete feature - and we haven't got close on achieving it.
- **Used `Cmp` instead of `CmpOrArgs` as the base generic type parameter to simplify** the type system in this addon project
- use legacy syntax on tests and components related to legacy API
- replace `#` imports _(which uses map from `pkg.imports`)_ inside `./src/runtime/**/*` - `svelte2tsx` doesn't support it (yet?) - ❤️ @paoloricciuti for [detailed investigation](https://discord.com/channels/486522875931656193/1298412930089615360)

<details>
  <summary>📦 Published PR as canary version: <code>4.1.8--canary.219.161a0fb.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-svelte-csf@4.1.8--canary.219.161a0fb.0
  # or 
  yarn add @storybook/addon-svelte-csf@4.1.8--canary.219.161a0fb.0
  ```
</details>

### CODE REVIEW

**Comment by xeho91** [comment]
> replace # imports (which uses map from pkg.imports) inside ./src/runtime/**/* - svelte2tsx doesn't support it (yet?) - ❤️ @paoloricciuti for [detailed investigation](https://discord.com/channels/486522875931656193/1298412930089615360)

@paoloricciuti Should we create an issue on [sveltejs/language-tools](https://github.com/sveltejs/language-tools) regarding this one?

**Comment by paoloricciuti** [comment]
> > replace # imports (which uses map from pkg.imports) inside ./src/runtime/**/* - svelte2tsx doesn't support it (yet?) - ❤️ @paoloricciuti for [detailed investigation](https://discord.com/channels/486522875931656193/1298412930089615360)
> 
> @paoloricciuti Should we create an issue on [sveltejs/language-tools](https://github.com/sveltejs/language-tools) regarding this one?

I'll investigate a bit better but from what I've seen the transformation is actually correct...the type is lost when TS emit the DTS.

File: examples/Button.stories.svelte
Comment (JReinhold) [inline examples/Button.stories.svelte:23]: I don't think this is a good example we want to document, and I don't think it's needed because of line 42 below?

If it's to satisfy types, then I think we should use type modifications instead like `as`.

File: src/index.test.ts
Comment (JReinhold) [inline src/index.test.ts:25]: same comment, this shouldn't be necessary

File: src/runtime/LegacyStory.svelte
Comment (JReinhold) [inline src/runtime/LegacyStory.svelte:23]: I don't understand this change TBH

File: src/runtime/Story.svelte
Comment (JReinhold) [inline src/runtime/Story.svelte:46]: These aren't correct, are they? We don't support `autodocs` and `source` attributes in the new `Story`, only in `LegacyStory`? I can't remember why we had them as type `never` instead of just removing them completely though. the "deprecated" tag is misleading, since these are actually removed, not just deprecated.

**Review by JReinhold — CHANGES_REQUESTED** [review]

File: src/runtime/Story.svelte
Comment (xeho91) [reply src/runtime/Story.svelte:46]: Yes, they're removed from the modern `Story` component.

The opt-in legacy support option handles them in `LegacyStory` with pre-transform plugin.

The reason I brought them back was to prevent users from having waterfall of type errors caused by usage of these props. It would occur while upgrading this addon in their existing project(s) to the next major version.

Hmm... Now given the second thought reading my reply, I should've put these type changes inside the `LegacyStory` instead. That sounds reasonable to you?

File: src/runtime/LegacyStory.svelte
Comment (xeho91) [reply src/runtime/LegacyStory.svelte:23]: Apparently Svelte language tools are so smart and capable to recognize when legacy API is used.
And producing two different types (via import `Story|LegacyStory.svelte`.

While using `LegacyStory`, the type of this component was treated as modern one (`svelte@5` - `Component` - function).
Hence props types we're mismatched and couldn't use let:*` directives without type errors.
Examples of these errors can be found in the other [PR comment](https://github.com/storybookjs/addon-svelte-csf/pull/215#issuecomment-2418701326)  (which can be closed if we merge this one)


So, for language tools to treat this component as legacy `SvelteComponent` (class), I had to use the legacy syntax for defining props.

File: examples/Button.stories.svelte
Comment (xeho91) [reply examples/Button.stories.svelte:23]: Changing to:

```diff
- children: createRawSnippet(() => ({ render: () => 'Click me' })),
+ children: 'Click me' as Snippet,
```

will produce the error:

```txt
Conversion of type 'string' to type 'Snippet<[]>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first. [2352]
```

So, the workaround is:

```diff
- children: 'Click me' as Snippet,
+ children: 'Click me' as unknown as Snippet,
```

Which is cumbersome. Shall we proceed with this?

Comment (xeho91) [reply examples/Button.stories.svelte]: I provided another alternative, which can be reverted: 104d5e0cb9dc818c104834f31755e44eb59b4f41

Comment (paoloricciuti) [reply examples/Button.stories.svelte]: Wait but do you want a snippet or a string? Because those are not interchangeable

Comment (JReinhold) [reply examples/Button.stories.svelte]: Well, we want a string here, because we convert it to a snippet down in the template below:

https://github.com/storybookjs/addon-svelte-csf/pull/219/files/6b646f1b2e9a44c8e52814b23ae2d71f93797b07#diff-27706a42acb167034a3d3aa2f621b4a43e00b1bd177c0e87c2028fad8e14b398R41-R43

This procedure ensures you can easily modify the arg as a string in the Control panel, while it still being passed as a snippet to the component.
But it's slightly cumbersome, and I haven't found a way yet to support this pretty common use case better yet - ideas welcome!

Comment (JReinhold) [reply examples/Button.stories.svelte]: @xeho91 your alternative looks cleaner, but now it doesn't allow you to modify it in Controls anymore though. but let's stick with it. 👍

File: src/runtime/LegacyStory.svelte
Comment (JReinhold) [reply src/runtime/LegacyStory.svelte]: oh damn that's good. great find! ❤️

File: src/runtime/Story.svelte
Comment (JReinhold) [reply src/runtime/Story.svelte]: hmm. 

> The reason I brought them back was to prevent users being flooded with type errors caused by usage of these props. It would occur while upgrading this addon in their existing project(s) to the next major version.

This seems reasonable to me.

> Hmm... Now given the second thought reading my reply, I should've put these type changes inside the LegacyStory instead. That sounds reasonable to you?

And this is also what would make sense to me - but doing this means that users will still get flooded with type errors, right? because no user is explicitly importing LegacyStory? I'm conflicted here. We could document that to get rid of these type errors, users need to do a search and replace for `Story -> LegacyStory` in their code base? But I'm not sure that even works at runtime, does it?

I'll defer to your best judgement here, I trust you.

**Review by JReinhold — APPROVED** [review]
Still have comments, but I'm approving, and letting you decide if you want to act on them or not. 🙂

File: src/runtime/Story.svelte
Comment (xeho91) [reply src/runtime/Story.svelte]: > And this is also what would make sense to me - but doing this means that users will still get flooded with type errors, right? **because no user is explicitly importing LegacyStory**? I'm conflicted here. We could document that to get rid of these type errors, users need to do a search and replace for `Story -> LegacyStory` in their code base? But I'm not sure that even works at runtime, does it?

Hm, I'm trying to get how you see that they could be flooded by those type errors? I'm guessing there's a confusion between `Story` and `LegacyStory` _(which is exported and aliased as `Story` for legacy compatibility)_.

Below is legacy syntax:

```svelte
<script>
  import { Story } from '@storybook/addon-svelte-csf';
  //       👆 This one is the `LegacyStory` now, so its old props are left untouched, except for the deprecation remarks.
</script>
```

Comment (JReinhold) [reply src/runtime/Story.svelte]: > which is exported and aliased as Story for legacy compatibility

Sorry, I totally forgot about that part. 😵‍💫

**Comment by shilman** [comment]
:rocket: PR was released in [`v5.0.0-next.11`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v5.0.0-next.11) :rocket:

**Pull request merged**
